src/rustdoc-internals/rustdoc-test-suite.md

```diff
@@ -0,0 +1,111 @@
+# The `rustdoc` test suite
+
+This page is specifically about the test suite named `rustdoc`, for other test suites used for testing rustdoc, see [Rustdoc tests](../rustdoc.md#tests).
+
+The `rustdoc` test suite is specifically used to test the HTML output of rustdoc.
+
+This is achived by means of `htmldocck.py`, a custom checker script that leverages [XPath].
+
+[XPath]: https://en.wikipedia.org/wiki/XPath
+
+## Directives
+Directives to htmldocck are similar to those given to `compiletest` in that they take the form of `//@` comments.
+
+In addition to the directives listed here,
+`rustdoc` tests also support most
+[compiletest directives](../tests/directives.html).
+
+All `PATH`s in directives are relative to the the rustdoc output directory (`build/TARGET/test/rustdoc/TESTNAME`),
```

Comment:
Unless I'm confusing myself, if you say the path is relative to `build/$TARGET/test/rustdoc/$CRATE_NAME` I would think a path like `foo/fn.bar.html` would expand to `build/$TARGET/test/rustdoc/foo/foo/fn.bar.html` assuming the crate name is `foo` (notice the double `foo/`).

I would describe it as relative to `build/$TARGET/test/rustdoc/` in which case this sentence merely describes an implementation detail of compiletest which the test writer shouldn't need to worry about.

Reply:
Is `tests/rustdoc` supposed to support revisions?

Reply:
They're currently rejected by compiletest (via an assert) since htmldocck.py has no notion of  that